src/Symfony/Component/HttpKernel/Log/DebugLoggerConfigurator.php

```diff
@@ -18,20 +18,21 @@
  */
 class DebugLoggerConfigurator
 {
-    private ?DebugLoggerInterface $processor = null;
+    private static \WeakMap $registeredLoggers;
+    private ?DebugLoggerInterface $debugLogger = null;
 
-    public function __construct(DebugLoggerInterface $processor, bool $enable = null)
+    public function __construct(DebugLoggerInterface $debugLogger, bool $enable = null)
     {
+        self::$registeredLoggers ??= new \WeakMap();
+
         if ($enable ?? !\in_array(\PHP_SAPI, ['cli', 'phpdbg', 'embed'], true)) {
-            $this->processor = $processor;
+            $this->debugLogger = $debugLogger;
         }
     }
 
-    public function pushDebugLogger(Logger $logger): void
+    public function registerForDebugLogger(Logger $logger): void
     {
-        if ($this->processor) {
-            $logger->pushProcessor($this->processor);
```

Comment:
If you are not pushing the processor in the monolog logger, it won't be receiving logs anymore.